lib/taskbuilder.js

```diff
@@ -90,8 +93,27 @@ function getPath(ops) {
  * @returns {Promise} - contains JSON object with the metadata
  */
 function getMetadata(iaName) {
+    return getAllMetadataFromServer()
+        .then(function (allMetadata) {
+            var iaData = allMetadata[iaName];
+
+            if (!iaData) {
+                return Promise.reject(new Error("Couldn't find metadata for IA: " + iaName + ". Is it live?"));
+            } else if (iaData.dev_milestone !== "live") {
+                return Promise.reject(new Error("IA " + iaName + " isn't live"));
+            } else {
+                return Promise.resolve(iaData);
+            }
+        });
+}
+
+function getAllMetadataFromServer() {
+    if (cachedMetadata) {
+        return Promise.resolve(cachedMetadata);
+    }
+
     return new Promise(function (resolve, reject) {
-        https.get("https://duck.co/ia/view/" + iaName + "/json", function (res) {
+        https.get("https://duck.co/ia/repo/all/json", function (res) {
```

Comment:
We should probably use the metadata file that lives on S3 here to prevent possible DOS? I think our other scripts use that instead of hitting the CommPlat directly.